### ISSUE TITLE
Ash

### ISSUE THREAD

**Issue by ndrean** [dwyl/learn-elixir#188]
Any experience with [Ash](https://www.ash-hq.org/)?  This declarative approach seems **really** interesting.

**Comment by nelsonic** [comment]
@ndrean thanks for opening this issue to capture the research into https://github.com/ash-project/ash 👌 
We briefly looked at it a while back but forgot to open an issue. 
It looks very promising but I don't immediately see the benefit over a standard `Phoenix` + `LiveView` App ...

![image](https://user-images.githubusercontent.com/194400/200550508-0a683114-46ea-42d2-9690-0ca8c80d56a6.png)

![image](https://user-images.githubusercontent.com/194400/200551723-e074bc3b-c47c-4c91-be29-1a06bcc66511.png)

![image](https://user-images.githubusercontent.com/194400/200550755-fa6ad697-7e54-4e96-879d-91e607ee91f0.png)

![image](https://user-images.githubusercontent.com/194400/200550282-131ed938-6cb5-42d5-92c5-2bb877e1c046.png)


Need to make time to follow https://www.ash-hq.org/docs/guides/ash/latest/tutorials/get-started 👩‍💻 
to determine if it would be _useful_ to us 🤞 
***or*** if it creates an unnecessary abstraction layer 🤷‍♂️ 
that makes debugging more tedious. ⏳

**Comment by ndrean** [comment]
Made me think of `docker-compose` over the Docker CLI, or Ecto over SQL. If you just make queries, the Ecto DSL (especially de "macro" `|>` one) is another - verbose, not so usefull - tool to learn over SQL. But as soon as you start INSERT data, then the `changeset` and `multi` abstraction with `Transaction` is pretty neat I think. I will try the example too.

**Comment by nelsonic** [comment]
Yeah, I often feel that the **Ecto** Abstraction is annoying and less maintainable than **SQL** … but that appears to be counter to the consensus in the Elixir/Phoenix community. 💭

Again, very curious to investigate this. 👌

**Comment by ndrean** [comment]
<https://jon.hk/elixir/ash/a-gentle-primer-to-ash/>

To read, one day!

<img width="1215" alt="Screenshot 2023-10-26 at 15 46 22" src="https://github.com/dwyl/learn-elixir/assets/6793008/331d41fa-df7c-4896-9b40-df21bbace443">
